online-shop/src/Components/EditProductPageComponent/EditProductSmartView.tsx

```diff
@@ -0,0 +1,186 @@
+        props.dispatch(initialProductDetailsLoad())
+    },
+    updateTemporaryProductValues: (props: EditProductPageProps) => (event: any, newProdValues: IProduct) => {
+        props.dispatch(updateTemporaryProduct(newProdValues))
+    },
+    onApplyChangesClicked: (props: EditProductPageProps) => (event: any) => {
+        if (didFormInputValuesChanged(props)) {
+            if (areNewFormValuesOk(props.temporaryProduct.name, props.temporaryProduct.image, props.temporaryProduct.description, props.temporaryProduct.price.toString())) {
+                const updatedProduct: IProduct = {
+                    id: props.productInEditStage.id,
+                    name: props.temporaryProduct.name,
+                    category: props.temporaryProduct.category,
+                    price: props.temporaryProduct.price,
+                    image: props.temporaryProduct.image,
+                    description: props.temporaryProduct.description
+                };
+
+                props.dispatch({ type: EDIT_PRODUCT, payload: updatedProduct })
+            }
+        } else {
+            alert("New values are needed for input fields to apply the changes.");
+        }
+    },
+    onTitleChange: (props: EditProductPageProps) => (event: any, value: string) => {
+        let title: string = value;
+
+        let auxTempObject: IProduct = cloneDeep(props.temporaryProduct);
+        auxTempObject.name = title;
+
+        props.dispatch(updateTemporaryProduct(auxTempObject));
+    },
+
+    onCategoryChange: (props: EditProductPageProps) => (event: any, value: string) => {
+        let category: string = value;
+
+        let auxTempObject: IProduct = cloneDeep(props.temporaryProduct);
+        auxTempObject.category = category;
+
+        props.dispatch(updateTemporaryProduct(auxTempObject));
+    },
+
+    onPriceChange: (props: EditProductPageProps) => (event: any, value: string) => {
+        let price: number = +value;  //string -> number conversion : +string = number
+
+        let auxTempObject: IProduct = cloneDeep(props.temporaryProduct);
+        auxTempObject.price = price;
+
+        props.dispatch(updateTemporaryProduct(auxTempObject));
+    },
+
+    onImageChange: (props: EditProductPageProps) => (event: any, value: string) => {
+        let image: string = value;
+
+        let auxTempObject: IProduct = cloneDeep(props.temporaryProduct);
+        auxTempObject.image = image;
+
+        props.dispatch(updateTemporaryProduct(auxTempObject));
+    },
+
+    onDescriptionChange: (props: EditProductPageProps) => (event: any, value: string) => {
+        let description : string = value.trim();
+        console.log(description);
+
+        let auxTempObject : IProduct = cloneDeep(props.temporaryProduct);
+        auxTempObject.description = description;
+
+        props.dispatch(updateTemporaryProduct(auxTempObject));
+    }
+})
+
+const onComponentDidMount = lifecycle<EditProductPageProps, {}, {}>({
```

Comment:
You could merge the two `lifecycle` calls into a single one (you can have multiple lifecycle methods in a single call).